parity-reconstruct/fuzz/fuzz_targets/random_decode.rs

```diff
@@ -0,0 +1,126 @@
+    data: Vec<[u8; BLOCKSIZE]>,
+    used: Vec<bool>,
+}
+
+impl<const BLOCKSIZE: usize> BlockStorage<BLOCKSIZE> {
+    fn new(n: usize) -> Self {
+        Self {
+            data: repeat([0; BLOCKSIZE]).take(n).collect(),
+            used: repeat(false).take(n).collect(),
+        }
+    }
+}
+
+struct WrappedBlockStorage<const BLOCKSIZE: usize>(Rc<RefCell<BlockStorage<BLOCKSIZE>>>);
+
+impl<const BLOCKSIZE: usize> DataStorage<BLOCKSIZE> for WrappedBlockStorage<BLOCKSIZE> {
+    fn store(&mut self, m: usize, data: [u8; BLOCKSIZE]) {
+        let mut this = self.0.borrow_mut();
+        assert!(!this.used[m]);
+        this.data[m] = data;
+        this.used[m] = true;
+    }
+
+    fn get(&self, m: usize) -> [u8; BLOCKSIZE] {
+        let this = self.0.borrow();
+        assert!(this.used[m]);
+        this.data[m]
+    }
+}
+
+impl<const BLOCKSIZE: usize> ParityStorage<BLOCKSIZE> for BlockStorage<BLOCKSIZE> {
+    fn store(&mut self, m: usize, data: [u8; BLOCKSIZE]) {
+        assert!(!self.used[m]);
+        self.data[m] = data;
+        self.used[m] = true;
+    }
+
+    fn get(&self, m: usize) -> [u8; BLOCKSIZE] {
+        assert!(self.used[m]);
+        self.data[m]
+    }
+}
+
+fuzz_target!(|testcase: (Vec<[u8;4]>, &[u8])| {
+    let data = &testcase.0;
+    let received: &BitSlice<_, Lsb0> = BitSlice::from_slice(testcase.1);
+    if data.len() < 2 || data.len() > 1024 || data.len() + 16 > received.count_ones() {
+        return;
+    }
+
+    let matrix = LfdbtParity::new(data.len());
+    let datastore = Rc::new(RefCell::new(BlockStorage::new(data.len())));
+    let mut reconstructor = Reconstructor::<_, _, _, _, 4, [usize; 1024/size_of::<usize>()], [usize; 1024/size_of::<usize>()] >::new(
+        data.len(),
+        LfdbtParity::new(data.len()),
+        TestMatrixStorage::new(data.len()),
+        BlockStorage::new(data.len()),
+        WrappedBlockStorage(datastore.clone()),
+    );
+
+    let mut total: BitArray<[usize; 1024/size_of::<usize>()]> = BitArray::ZERO;
+    for (i, provide) in received.iter().enumerate() {
+        if *provide {
+            let row: BitArray<[usize; 1024/size_of::<usize>()]> = matrix.row(i);
+            total |= row;
+            let mut block = [0;4];
+            for (j, take) in row.iter().enumerate().take(data.len()) {
+                if *take {
+                    for (blockel, datael) in block.iter_mut().zip(data[j].iter()) {
+                        *blockel ^= *datael
+                    }
+                }
+            }
+
+            match reconstructor.handle_block(i, block) {
+                BlockResult::NeedMore => {},
+                BlockResult::TooManyMissing => panic!("Should always fit"),
+                BlockResult::Done => {
+                    let store = datastore.borrow();
+                    assert_eq!(&store.data, data);
+                    assert!(!store.used.iter().any(|v| !*v));
+                    return;
+                }
+            }
+        }
+    }
```

Comment:
It's not super clear what we're doing here and what our test goal is. Some extra comments would go a long way.

Also, does this simulate packet loss? I see the `if *provide`, but I'm not sure what this does